src/language-js/print/type-parameters.js

```diff
@@ -152,7 +152,11 @@ function printTypeParameter(path, options, print) {
   parts.push(name);
 
   if (node.bound) {
-    parts.push(printTypeAnnotationProperty(path, print, "bound"));
+    if (node.usesExtendsBound) {
+      parts.push(" extends ", print(["bound", "typeAnnotation"]));
```

Comment:
`node.bound` is skipped, better not to do that, there could be comments, cursors.

This won't work?

```js
parts.push(" extends ", print("bound"));
```

Reply:
Ah, I understand the problem, we should still use `printTypeAnnotationProperty`. I'll fix it and add comments to explain why it's needed and how it works on monday.

Reply:
Found a case

**Prettier pr-14573**
[Playground link](https://deploy-preview-14573--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuEMCeAHOACAgtgXmwAoBKQgPmwEsoAzOAJ2wCVs4APeKAEwGcA9ACpskALZiEMbEIHY+MRrQDmAHSggANCAgYY1aH2SgAho0YQA7gAUzCIyhMAbSybRHtAI0YmwAazgYAGUTSQAZWjhkOmc+OC8ff0CgjF8VZEUAV3iQODFPOB4eQrCTKGVMk2U4ADEIRjETGH1y5BATTJgILRAACxgxJwB1Xup4PlSwOCD7MeoANzG0NrA+DxBaOMYYax9lRujYnIArPg4glSc4AEVMiHhDpzjtVMYttronKx6MJVgh6g8GC9ZAADgADC8LHEhj4MG1fnAtvMotoAI53eC7XQOdp8AC0UDghUKPUYcAx1HJuyqByQMSeOTiYmoGUY2W0fEucFwzSUnk6cGsTAiRMezxAXPKV1u9yi9KO2hgJk8AKBIKQACYlT5qE4VABhCASExtZxOHqZOIAFRVDgZEvm2QAkrwpEEwEo9LheEF0FdxXAAL5BoA)
<!-- prettier-ignore -->
```sh
--parser flow
```

**Input:**
<!-- prettier-ignore -->
```jsx
type A = () => infer R extends/* comment */ string

```

**Output:**
<!-- prettier-ignore -->
```jsx
Error: Comment "comment" was not printed. Please report this error!
    at dn (https://deploy-preview-14573--prettier.netlify.app/lib/standalone.js:29:5817)
    at Ze (https://deploy-preview-14573--prettier.netlify.app/lib/standalone.js:29:7308)
    at async kn (https://deploy-preview-14573--prettier.netlify.app/lib/standalone.js:29:11045)
    at async ju (https://deploy-preview-14573--prettier.netlify.app/lib/standalone.js:34:581)
    at async formatCode (https://deploy-preview-14573--prettier.netlify.app/worker.js:159:12)
    at async handleFormatMessage (https://deploy-preview-14573--prettier.netlify.app/worker.js:83:24)
    at async self.onmessage (https://deploy-preview-14573--prettier.netlify.app/worker.js:38:14)
```